src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py

```diff
@@ -0,0 +1,269 @@
+
+    def configure_terraform(self, tf_folder, num_worker_nodes, api_vip_ip):
+        tfvars = utils.get_tfvars(tf_folder)
+        self.configure_terraform_workers_nodes(tfvars, num_worker_nodes)
+        tfvars["api_vip"] = api_vip_ip
+        utils.set_tfvars(tf_folder, tfvars)
+
+    def configure_terraform_workers_nodes(self, tfvars, num_worker_nodes):
+        num_workers = tfvars["worker_count"] + num_worker_nodes
+        tfvars["worker_count"] = num_workers
+        self.set_workers_addresses_by_type(
+            tfvars, num_worker_nodes, "libvirt_master_ips", "libvirt_worker_ips", "libvirt_worker_macs"
+        )
+        self.set_workers_addresses_by_type(
+            tfvars,
+            num_worker_nodes,
+            "libvirt_secondary_master_ips",
+            "libvirt_secondary_worker_ips",
+            "libvirt_secondary_worker_macs",
+        )
+
+    def set_workers_addresses_by_type(self, tfvars, num_worker_nodes, master_ip_type, worker_ip_type, worker_mac_type):
+        old_worker_ips_list = tfvars[worker_ip_type]
+        last_master_addresses = tfvars[master_ip_type][-1]
+
+        if last_master_addresses:
+            if old_worker_ips_list:
+                worker_starting_ip = ipaddress.ip_address(old_worker_ips_list[-1][0])
+            else:
+                worker_starting_ip = ipaddress.ip_address(last_master_addresses[0])
+
+            worker_ips_list = old_worker_ips_list + utils.create_ip_address_nested_list(
+                num_worker_nodes, worker_starting_ip + 1
+            )
+        else:
+            log.info(
+                "IPv6-only environment. IP addresses are left empty and will be allocated by libvirt "
+                "DHCP because of a bug in Terraform plugin"
+            )
+            worker_ips_list = old_worker_ips_list + utils.create_empty_nested_list(num_worker_nodes)
+
+        tfvars[worker_ip_type] = worker_ips_list
+
+        old_worker_mac_addresses = tfvars[worker_mac_type]
+        tfvars[worker_mac_type] = old_worker_mac_addresses + static_network.generate_macs(num_worker_nodes)
+
+    def get_ocp_cluster_nodes(self, kubeconfig):
+        res = subprocess.check_output(f"oc --kubeconfig={kubeconfig} get nodes --output=json", shell=True)
+        return json.loads(res)["items"]
+
+    def is_ocp_node_ready(self, node_status):
+        if not node_status:
+            return False
+        for condition in node_status["conditions"]:
+            if condition["status"] == "True" and condition["type"] == "Ready":
+                return True
+        return False
+
+    def are_libvirt_nodes_in_cluster_hosts(self, client, cluster_id, num_nodes):
+        try:
+            hosts_macs = client.get_hosts_id_with_macs(cluster_id)
+        except BaseException:
+            log.error("Failed to get nodes macs for cluster: %s", cluster_id)
+            return False
+        num_macs = len([mac for mac in hosts_macs if mac != ""])
+        return num_macs >= num_nodes
```

Comment:
why not using `self._nodes.nodes_count`?

Reply:
it's the day2 nodes, can use self.config.num_day2_workers